quelpa.el

```diff
@@ -1094,7 +1113,17 @@ Optionally PRETTY-PRINT the data."
 
 (defun quelpa-build--create-tar (file dir &optional files)
   "Create a tar FILE containing the contents of DIR, or just FILES if non-nil."
-  (when (eq system-type 'windows-nt)
+  (when (and (not quelpa-tar-executable-type)
```

Comment:
Thanks, this looks much better.

If I may make one more suggestion, I think it would be helpful to put this code in a function.  e.g.

```el
(defun quelpa--tar-type ()
  "Return `bsd' or `gnu' depending on type of Tar executable.
Tests and sets `quelpa-tar-executable-type' if not already set."
  (or quelpa-tar-executable-type
      (when (and quelpa-build-tar-executable
                 (file-executable-p quelpa-build-tar-executable))
        (setq quelpa-tar-executable-type
              (let ((v (shell-command-to-string
                        (format "%s --version" quelpa-build-tar-executable))))
                (cond ((string-match-p "bsdtar" v) 'bsd)
                      ((string-match-p "GNU tar" v) 'gnu)
                      (t 'gnu)))))))
```

Reply:
Done